src/tdigest.rs

```diff
@@ -279,20 +207,16 @@ fn tdigest_final(
 
             // we need to flatten the vector to a single buffer that contains
             // both the size, the data, and the varlen header
-            let flattened = flatten! {
-                TsTDigestData{
-                    header: &0,
-                    buckets: &buckets,
-                    count: &count,
-                    sum: &state.digested.sum(),
-                    min: &state.digested.min(),
-                    max: &state.digested.max(),
-                    means: &means,
-                    weights: &weights,
-                }
-            };
-
-            TimescaleTDigest(flattened).into()
+            TsTDigestData {
+                header: &0,
```

Comment:
Not great that we need an initialization for this never seen before or again field here.  Another argument for using another macro here, if that approach is possible.